chirps/scan/views.py

```diff
@@ -33,11 +32,73 @@ def result_detail(request, scan_id, policy_id, rule_id):
     return render(request, 'scan/result_detail.html', {'results': results, 'findings': findings})
 
 
+@login_required
+def view_scan(request, scan_id):
+    """View details for a particular scan."""
+    scan = get_object_or_404(Scan, pk=scan_id, user=request.user)
+
+    # Assemble a slit of all the results that had findings, across all targets
+    results = []
+    scan_targets = ScanTarget.objects.filter(scan=scan)
+
+    # Step 1: build a list of all the results (rules) with findings.
+    for scan_target in scan_targets:
+        # Iterate through the rule set
+        for result in scan_target.results.all():
+            if result.has_findings():
+                results.append(result)
+
+    # Step 2: aggregate results by rules with matching rule IDs
+    unique_rules = []
+
+    # First, build a list of all the unique rule IDs
+    for result in results:
+        if result.rule not in unique_rules:
+            unique_rules.append(result.rule)
+
+    # Next, walk through all of the results, aggregating the findings count for each unique rule ID
+    finding_count = 0
+    finding_severities = {}
+    for rule in unique_rules:
+        rule.finding_count = 0
+        rule.findings = []
+        for result in results:
+
+            # Increment the number of findings for this rule
+            if result.rule.id == rule.id:
+                findings = result.findings.all()
+                count = findings.count()
+                rule.finding_count += count
+                finding_count += count
+                rule.findings.extend(list(findings))
+
+                # While we're in this loop, store off the number of times each severity is encountered
+                # This will be used to render the pie-chart in the UI
+                if rule.severity not in finding_severities:
+                    finding_severities[rule.severity] = 0
+
+                finding_severities[rule.severity] += count
```

Comment:
You might be able to simplify the logic here with the use of `collections.defaultdict`

```suggestion
from collections import defaultdict

finding_count = 0
finding_severities = defaultdict(int)

for rule in unique_rules:
    rule.finding_count = 0
    rule.findings = []
    for result in results:
        if result.rule.id == rule.id:
            findings = list(result.findings.all())
            count = len(findings)
            rule.finding_count += count
            finding_count += count
            rule.findings.extend(findings)
            finding_severities[rule.severity] += count
```